### DISCUSSION TITLE
OPC UA issues Write to tags

### DISCUSSION THREAD

**Discussion by LAB-KN** [frangoteam/FUXA#1323] [Q&A] [answered]
I have setup a OPC UA in my Siemens S7-1200 PLC.
I can access the server interface, and can read all the tags.

When I add a tag to my hmi (with Input Control), and change the value, it is not written to the OPC.
I can connect to the server with other software, and can write to the tag no problem.

Is there any setup i need to do for OPC Write to be enabled?

**Comment by LAB-KN** [comment]
It seems like i only have issue when the OPC UA is not showing a data type.
When i test with a datatype String it can only read the value, and Type is empty
If i test a datatype Int, it can read and write no problem.


![image](https://github.com/user-attachments/assets/7396c074-578d-4293-94fd-d4e196bbcb9a)


Edit:
Reading the info with other software shows the type as "String"
![image](https://github.com/user-attachments/assets/aebb9b87-aa06-45ff-b8f6-bdc72dbff0ed)

**Reply by LAB-KN** [reply]
I seem to have found a workaround.
I'm not sure if this is considered a bug, or working as intended.

If i setup the Node Type as a wstring on the PLC side, Fuxa can see it as a String.

![image](https://github.com/user-attachments/assets/cf10eabf-6038-4812-ac43-0a3abf90d8a2)

**Reply by unocelli** [reply, marked as answer]
Hi, please add it to issues
